sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -151,16 +151,63 @@ def fit(self, X, y):
         return self
 
 
-class NoSparseClassifier(BaseBadClassifier):
-    def fit(self, X, y):
-        X, y = check_X_y(X, y, accept_sparse=['csr', 'csc'])
+class NoSparseClassifierException(LogisticRegression):
```

Comment:
I made the fake estimators inherit from `LogisticRegression` so that they pass all the `check_estimator` checks, (except the one except the one we want to test like `check_estimator_sparse_data` for NoSparseClassifier, `NoSparseClassifierTypeError` or `NoSparseClassifierValueError`, or `check_estimator_sparse_dense` for `SparseDenseDifferentPredict` or `SparseDenseDifferentPredictProba`)

Reply:
(Note that there was also `check_estimator_unfitted` that used `NoSparseClassifier`, but I made it use a plain `BaseBadClassifier` instead so that it indeed uses a Classifier that doesn't raise an error message when not fitted)

Reply:
I think it would be enough to derive from `BaseBadClassifier`. This is only meant to check that the estimator checks that checks the error message is informative right? Let's keep it as simple as possible.

Reply:
Also the name `NoSparseClassifierException` is weird because one my except it to be a exception class (a subclass of `Exception`).

Instead I would name it something like `BadFitExceptionSparseClassifier` or something.

Reply:
Originally I inherited from `LogisticRegression` so that no error, other than the one I wanted, was raised (I just wanted `Exception("Nonsensical Error")` to be raised) . But I agree it's a bit confusing, so instead I falled back to inheriting from `BaseBadClassifier` as you said and just ensured to do the necessary checks so that no unwanted error is thrown (see  https://github.com/scikit-learn/scikit-learn/pull/13246/commits/70e9d2c9bc103f4bc734789e376706c7cabb92f3)

Reply:
I also changed the name to `BadFitExceptionSparseClassifier `